vernac/debugHook.mli

```diff
@@ -66,26 +67,28 @@ module Answer : sig
     | Prompt of Pp.t (* output signalling the debugger has stopped
                         Should be printed as a prompt for user input,
                         e.g. in color without a newline at the end *)
-    | Goal of Pp.t   (* goal for the current proof state *)
     | Output of Pp.t (* general output *)
     | Init           (* signals initialization of the debugger *)
-    | Stack of (string * (string * int list) option) list
```

Comment:
I'd be good if we don't need to depend on the `protocol` which is specific to CoqIDE, whereas this interface is generic. Protocol can instead use the types here.

Reply:
Updated. I created a new file with the needed types.  I wasn't sure where to put it; `lib` was convenient.  I'm not keen on adding new module dependencies for small changes, or creating new files that contain only a single definition--that seems like the path to clutter.  Where would you put it?

Reply:
I'd say it belongs in `debugHook`

Reply:
OK.  `debugHook` is in the `vernac` module, which isn't and shouldn't be linked into CoqIDE.  Where do you suggest moving `debugHook`?

Reply:
Indeed we could introduce a library for RPC types, not sure it is worth it for just this type tho. A simple solution could be (only for now) to duplicate the type (and add a comment)

Reply:
Unfortunately, AFAIK, OCaml won't consider the duplicated types as equivalent without resorting to conversion functions such as:

```
external convertAB : A.t -> B.t = "%identity"
external convertBA : B.t -> A.t = "%identity"
```

a technique that I expect we don't want to use.  The last commit I just pushed fails with an emblematic error:

```
File "ide/coqide/idetop.ml", line 536, characters 22-41:
536 |     Interface.goals = interruptible goals;
                            ^^^^^^^^^^^^^^^^^^^
Error: This expression has type
         Interface.goals_sty -> DebugHook.goal DebugHook.pre_goals option
       but an expression was expected of type
         Interface.goals_sty -> Interface.goals_rty
       Type DebugHook.goal DebugHook.pre_goals option
       is not compatible with type
         Interface.goals_rty = DebuggerTypes.goals option
       Type DebugHook.goal DebugHook.pre_goals is not compatible with type
         DebuggerTypes.goals = DebuggerTypes.goal DebuggerTypes.pre_goals
```

I think our only choice is where to put debuggerTypes.ml :-(.